website/cue/reference/releases/0.29.0.cue

```diff
@@ -0,0 +1,594 @@
+			scopes: ["webhdfs sink"]
+			description: """
+				A new `webhdfs` sink was added to send log data to HDFS via the
+				[WebHDFS](https://hadoop.apache.org/docs/r1.0.4/webhdfs.html) interface.
+
+				This sink is also the first to use the new
+				[OpenDAL](https://github.com/apache/incubator-opendal) library which may enable us
+				to more quickly add support for other sinks in the future as that library expands
+				storage support.
+				"""
+			contributors: ["Xuanwo"]
+			pr_numbers: [16557]
+		},
+		{
+			type: "fix"
+			scopes: ["aws provider"]
+			description: """
+				Support for AWS credential files was re-added for all AWS components. This was
+				dropped during the migration of the SDK we were using from `rusoto` to AWS's
+				official Rust SDK.
+				"""
+			contributors: ["protochron"]
+			pr_numbers: [16633]
+		},
+		{
+			type: "enhancement"
+			scopes: ["loki sink"]
+			description: """
+				The `loki` sink now supports sending a field containing a map as labels with
+				configuration like:
+
+				```yaml
+				labels:
+					"*": "{{ labels }}"
+				```
+
+				to send the key/value map stored at `labels` as the set of labels to Loki.
+				"""
+			contributors: ["hargut"]
+			pr_numbers: [16591]
+		},
+		{
+			type: "enhancement"
+			scopes: ["aws provider"]
+			description: """
+				AWS components can now also assume-role whenever static credentials are provided in the
+				configuration file through the `auth.assume_role` option.
+				"""
+			contributors: ["sbalmos"]
+			pr_numbers: [16715]
+		},
+		{
+			type: "feat"
+			scopes: ["codecs", "sinks"]
+			description: """
+				A new `csv` encoding option was added for sinks. Example configuration:
+
+				```yaml
+				encoding:
+					codec: csv
+					csv:
+						fields:
+						- field1
+						- field2
+				```
+
+				This will encode the event as CSV using the `field1` and `field2` fields, in order.
+
+				Note that CSV headers are not added. This requires batch header support that will be
+				added in the future.
+				"""
+			contributors: ["everpcpc"]
+			pr_numbers: [16603, 16828]
+		},
+		{
+			type: "enhancement"
+			scopes: ["datadog provider"]
+			description: """
+				The deprecated `api_key` option was removed from Datadog sinks. It had been replaced
+				by `default_api_key`.
+				"""
+			breaking: true
+			pr_numbers: [16750]
+		},
+		{
+			type: "fix"
+			scopes: ["aws_s3 source"]
+			description: """
+				The `aws_s3` source no longer deadlocks under low `client_concurrency` when
+				`acknowledgements` are enabled.
+				"""
+			contributors: ["mdeltito"]
+			pr_numbers: [16742]
+		},
+		{
+			type: "chore"
+			scopes: ["socket source"]
+			description: """
+				The `socket` source `max_length` option was deprecated in-lieu of the options
+				available on `framing`.
+
+				See [the upgrade
+				guide](/highlights/2023-04-11-0-29-0-upgrade-guide#socket-source-max-length) for
+				more details.
+				"""
+			pr_numbers: [16752]
+		},
+		{
+			type: "fix"
```

Comment:
Probably isn't a fix.

Reply:
🤔 but if every bug is a feature is every feature a bug?

Thanks for catching this copy/pasta.